wasm/src/test/java/com/dylibso/chicory/wasm/WasmWriterTest.java

```diff
@@ -0,0 +1,31 @@
+package com.dylibso.chicory.wasm;
+
+import static com.dylibso.chicory.wasm.Parser.parseWithoutDecoding;
+import static com.dylibso.chicory.wasm.ParserTest.wasmCorpusFiles;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class WasmWriterTest {
+
+    @Test
+    public void shouldRoundTrip() throws IOException {
+        for (var file : wasmCorpusFiles()) {
+            // uses non-canonical size encodings
```

Comment:
What does this means? Can you elaborate?
Sounds more like a bug ...

Reply:
The `main.go.wasm` file encodes a section size as `F2 80 80 80 00` rather than the canonical `72`. When writing the output, we write the size using the canonical encoding, which means the round-trip test will fail as the bytes are different between the original and what we wrote.

Reply:
I didn't know about this and found:
https://github.com/WebAssembly/design/issues/533

That's surprising, but ok.